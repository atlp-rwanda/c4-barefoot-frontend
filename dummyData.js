export const user = [
  {
    "first_name":"M",
    "last_name":"JAckson",
    "email":"jackson.novelty@gmail.com",
    "occupation":"Programming",
    "username":"jacks",
    "bio":"Software Developer",
    "address":"kk 509 st",
    "language":"Kinyarwanda",
    "password":"12345678",
    "profile_picture":"https://res.cloudinary.com/mjackson/image/upload/v1609304524/Screen_Shot_2020-12-29_at_08.21.43_qe7pem.png"
  }
]

export const signupState = {
    requesting: false,
    errorOpen: false,
    erro: ""
  }
export const locationsPayload = [
    {
      id: "c6028e0d-ef88-4693-ab49-f37669891724",
      LocationName: "Kigali",
      country: "Rwanda",
      description: "The land of a thousand hills",
      link: "https://www.nationalgeographic.com/content/dam/travel/Guide-Pages/africa/downtown-kigali-rwanda.adapt.1900.1.jpg",
    }
  ]

export const accommodationsPayload = [
    {
      
        id: "0ce36391-2c08-3074-bddb-a4ea8cccbbc5",
        country: "Rwanda",
        city: "Kigali",
        title: "Marriot Hotel",
        description: "A serene environment for relaxation",
        photos: "https://cache.marriott.com/marriottassets/marriott/KGLMC/kglmc-exterior-0030-hor-feat.jpg",
    
    }
  ]

  export const accommodationState = {
    pending: false,
    accommodations: accommodationsPayload,
    error: null
  }

  export const locationState = {
    pending: false,
    locations: locationsPayload,
    error: null
  }

