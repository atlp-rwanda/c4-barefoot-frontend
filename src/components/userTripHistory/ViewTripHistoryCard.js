import moment from "moment";
import { connect } from "react-redux";
import React, { useEffect, useState } from "react";
import { useParams, Link } from "react-router-dom";
import Pagination from "@material-ui/lab/Pagination";
import { Button, Grid, makeStyles, Paper, Typography } from "@material-ui/core";
import { getTripHistory } from "../../redux/actions/userTravelHistoryAction";
import { getAccommodation } from "../../redux/actions/userTravelHistoryAction";

const useStyle = makeStyles(() => ({
  paper: {
    padding: 5,
    textAlign: "center",
    margin: "1em",
  },
  container: {
    margin: "2% 5% 2% 5%",

    width: "90%",
  },
  centered: {
    margin: "50px",
    fontSize: "2em",
    color: "blue",
  },
  media: {
    height: 140,
  },
  root: {
    maxWidth: 345,
  },
  media: {
    height: 140,
  },
  paganete: {
    marginTop: "20px",
    width: "100%",
    display: "flex",
    justifyContent: "center",
    marginBottom: "15px",
  },
}));

function ViewTripHistoryCard(props) {
  const [page, setPage] = useState(1);
  const page_size = 6;
  // const filtered = filter(travel, category);

  const handlePage = (e, value) => {
    setPage(value);
  };
  const { location } = props.match.params;
  let tripp = [];
  let trippp = [];

  useEffect(() => {
    props.getTripHistory();
    props.getAccommodation();
  }, []);

  const { trips, pending } = props.trips;
  trips.forEach((trip) => {
    tripp.push(trip.travelRequestInfo.Trip);
  });
  tripp.forEach((item) => {
    item.map((trip) => trippp.push(trip));
  });
  console.log(trippp.length);
  console.log(tripp);

  const classes = useStyle();
  const acc = props.acc;

  const paginate = (array, page_number) => {
    // human-readable page numbers usually start with 1, so we reduce 1 in the first argument
    return array.slice((page_number - 1) * page_size, page_number * page_size);
  };
  const paginated = paginate(trippp, page);

  return (
    <div>
      <Grid container spacing={1} className={classes.container}>
        {!pending &&
          paginated.map((trip) => (
            <Grid item key={trip.tripId} md={4} sm={6} xs={12}>
              <Paper className={classes.paper}>
                {acc && <img src={acc.photos} style={{ width: 150 }}></img>}
                <Typography>Destination: {trip.destination}</Typography>
                <Typography>Origin City: {trip.originCity}</Typography>
                <Typography>Reason: {trip.reason}</Typography>
                <Typography>
                  From: {moment(trip.tripDate).format("DD/MMM/YYYY")}
                </Typography>
                <Typography>
                  To: {moment(trip.returnDate).format("DD/MMM/YYYY")}
                </Typography>
                <Button
                  color="primary"
                  variant="contained"
                  onClick={() => {
                    localStorage.setItem("accId", trip.accommodationId);
                    localStorage.setItem("destination", trip.destination);
                    localStorage.setItem("origin", trip.originCity);
                    localStorage.setItem("reason", trip.reason);
                    localStorage.setItem("departure", trip.tripDate);
                    localStorage.setItem("returning", trip.returnDate);
                  }}
                >
                  <Link
                    to="/individual-history"
                    style={{ color: "white", textDecoration: "none" }}
                  >
                    Details
                  </Link>
                </Button>
              </Paper>
            </Grid>
          ))}
        {pending && (
          <Typography className={classes.centered}>Loading</Typography>
        )}
        {!pending && trippp.length == 0 && (
          <Typography className={classes.centered}>
            No trips available
          </Typography>
        )}
      </Grid>
      <div className={classes.paganete}>
        {trippp.length !== 0 && trippp.length / page_size > 1 && (
          <Pagination
            count={Math.ceil(trippp.length / page_size)}
            color="primary"
            onChange={handlePage}
          />
        )}
      </div>
    </div>
  );
}

const mapStateToProps = (state) => ({
  trips: state.tripHistory,
  locations: state.tripHistory.locations,
  acc: state.tripHistory.acc,
});
export { ViewTripHistoryCard };
export default connect(mapStateToProps, { getTripHistory, getAccommodation })(
  ViewTripHistoryCard
);
