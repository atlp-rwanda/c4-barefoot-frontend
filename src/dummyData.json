{
    "locations": [
      {
        "city": "capetown",
        "description": "This is one of best cities in Africa and you can also find good accomodations.",
        "image": "https://freewalkingtourscapetown.co.za/wp-content/uploads/2018/05/cropped-cape-town-header-pic-1.png"
      },
      {
        "city": "capetown",
        "description": "This is one of best cities in Africa and you can also find good accomodations.",
        "image": "https://freewalkingtourscapetown.co.za/wp-content/uploads/2018/05/cropped-cape-town-header-pic-1.png"
      },
      {
        "city": "capetown",
        "description": "This is one of best cities in Africa and you can also find good accomodations.",
        "image": "https://freewalkingtourscapetown.co.za/wp-content/uploads/2018/05/cropped-cape-town-header-pic-1.png"
      }
    ],
    "accommodations": [
      {
        "title": "Marriott Hotel",
        "location": "Kigali",
        "description": "This is one of best cities in Africa and you can also find good accomodations.",
        "image": "https://cache.marriott.com/marriottassets/marriott/KGLMC/kglmc-exterior-0030-hor-feat.jpg"
      },
      {
        "title": "Serena Hotel",
        "location": "Kigali",
        "description": "This is one of best cities in Africa and you can also find good accomodations.",
        "image": "https://cache.marriott.com/marriottassets/marriott/KGLMC/kglmc-exterior-0030-hor-feat.jpg"
      },
      {
        "title": "Marriott Hotel",
        "location": "Kigali",
        "description": "This is one of best cities in Africa and you can also find good accomodations.",
        "image": "https://cache.marriott.com/marriottassets/marriott/KGLMC/kglmc-exterior-0030-hor-feat.jpg"
      }
    ]
}
