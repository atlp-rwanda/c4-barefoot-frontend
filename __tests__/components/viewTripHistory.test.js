import React from 'react';
import { Skeleton } from '@material-ui/lab';
import { shallow, mount } from 'enzyme';
import { Provider } from "react-redux";
import configureStore from 'redux-mock-store';
import thunk from "redux-thunk";

import toJson from 'enzyme-to-json';
import ViewTripHistoryCard from '../../src/components/userTripHistory/ViewTripHistoryCard';
import { accommodationInfo, allTravelRequestState, travelRequestInfo, locationState, UserInfo } from '../../dummyData';

const middlewares = [thunk]
const mockStore = configureStore(middlewares);
let store;
describe(' <TravelRequestcard />', ()=>{
    let props;
    let wrapper;
   


    it( 'It should render the ReportsView compnent', ()=>{
        store = mockStore({
            trips: allTravelRequestState,
            acc: accommodationInfo,
            locations: locationState,
          });
    
        props= {
            userInfo: UserInfo,
            accomodationsInfo: accommodationInfo,
            // travel: travelRequestInfo,
            // category: 'approved',
            trips: allTravelRequestState,
            acc: accommodationInfo,
            locations: locationState,

            handleSingleTravel: jest.fn(),
               
        }
        wrapper= mount(<Provider store= {store} ><ViewTripHistoryCard { ...props } /></Provider>)
        const component= toJson(wrapper);
        expect(component).toMatchSnapshot();
        // console.log('component', component);
    })

   
    
})

