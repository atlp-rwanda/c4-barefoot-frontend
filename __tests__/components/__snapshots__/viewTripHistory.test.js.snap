// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` <ViewTripHistoryCard /> It should render the trip history compnent 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <ViewTripHistoryCard
    accomodationsInfo={
      Array [
        Object {
          "city": "Kigali",
          "country": "Rwanda",
          "description": "A serene environment for relaxation",
          "id": "598d90a3-68e6-43ac-970a-2855657be478",
          "locationID": "c6028e0d-ef88-4693-ab49-f37669891724",
          "numberOfRooms": 100,
          "photos": "image2.png",
          "propertyType": "Hostel",
          "state": "Kicukiro",
          "streetAddress": "KN 45 ST",
          "title": "Marina Hotel",
          "typeOfBed": "Double Decker",
        },
      ]
    }
    getAccommodation={[MockFunction]}
    getTripHistory={[MockFunction]}
    match={
      Object {
        "params": Object {
          "location": "Nyanza",
        },
      }
    }
    travel={
      Object {
        "Trip": Array [
          Object {
            "accommodationId": "598d90a3-68e6-43ac-970a-2855657be478",
            "createdAt": "2021-03-02T11:18:36.600Z",
            "destination": "KARONGI",
            "originCity": "MUSANZE",
            "reason": "Research and market development trip in rular areas",
            "returnDate": null,
            "travelId": "59ba77ce-3649-4b8a-bed6-b127cc1c530f",
            "tripDate": "2020-12-20T08:53:31.303Z",
            "tripId": "98624254-da4c-47bd-91da-53d9d89172cc",
            "updatedAt": "2021-03-02T11:18:36.600Z",
          },
        ],
        "createdAt": "2021-03-02T11:18:36.501Z",
        "managerId": "0ce36391-2c08-4703-bddb-a4ea8cccbbc5",
        "status": "approved",
        "travelId": "59ba77ce-3649-4b8a-bed6-b127cc1c530f",
        "updatedAt": "2021-03-02T11:21:00.554Z",
        "userId": "d74fcc5e-5755-4366-83ef-cf306b013c46",
      }
    }
    trips={
      Array [
        Object {
          "accommodationId": "598d90a3-68e6-43ac-970a-2855657be478",
          "createdAt": "2021-03-02T11:18:36.600Z",
          "destination": "KARONGI",
          "originCity": "MUSANZE",
          "reason": "Research and market development trip in rular areas",
          "returnDate": null,
          "travelId": "59ba77ce-3649-4b8a-bed6-b127cc1c530f",
          "tripDate": "2020-12-20T08:53:31.303Z",
          "tripId": "98624254-da4c-47bd-91da-53d9d89172cc",
          "updatedAt": "2021-03-02T11:18:36.600Z",
        },
      ]
    }
    userInfo={
      Object {
        "firstName": "pushRequester",
        "lastName": "One",
        "profilePpicture": "https://www.cobdoglaps.sa.edu.au/wp-content/uploads/2017/11/placeholder-profile-sq.jpg",
        "username": "pushrequester",
      }
    }
  />
</ContextProvider>
`;
